exercises/practice/bank-account/bank_account_test.cpp

```diff
@@ -0,0 +1,155 @@
+#include "bank_account.h"
+#ifdef EXERCISM_TEST_SUITE
+#include <catch2/catch.hpp>
+#else
+#include "test/catch.hpp"
+#endif
+#include <chrono>
+#include <string>
+#include <thread>
+
+TEST_CASE("newly_opened_account_has_zero_balance") {
+    Bankaccount account{};
+    account.open();
+    REQUIRE(account.balance() == 0);
+}
+
+#if defined(EXERCISM_RUN_ALL_TESTS)
+TEST_CASE("deposit_money_increases_balance") {
+    Bankaccount account{};
+    account.open();
+    account.deposit(100);
+    REQUIRE(account.balance() == 100);
+}
+
+TEST_CASE("deposit_money_sequentially_increases_balance") {
+    Bankaccount account{};
+    account.open();
+    account.deposit(100);
+    account.deposit(50);
+    REQUIRE(account.balance() == 150);
+}
+
+TEST_CASE("withdraw_money_decreases_balance") {
+    Bankaccount account{};
+    account.open();
+    account.deposit(100);
+    account.withdraw(50);
+    REQUIRE(account.balance() == 50);
+}
+
+TEST_CASE("withdraw_money_sequentially_decreases_balance") {
+    Bankaccount account{};
+    account.open();
+    account.deposit(100);
+    account.withdraw(50);
+    account.withdraw(30);
+    REQUIRE(account.balance() == 20);
+}
+
+TEST_CASE("checking_balance_of_not_opened_account_throws_error") {
+    Bankaccount account{};
+    REQUIRE_THROWS_AS(account.balance(), std::logic_error);
+}
+
+TEST_CASE("checking_balance_of_a_closed_account_throws_error") {
+    Bankaccount account{};
+    account.open();
+    account.close();
+
+    REQUIRE_THROWS_AS(account.balance(), std::logic_error);
+}
+
+TEST_CASE("deposit_into_closed_account_throws_error") {
+    Bankaccount account{};
+    account.open();
+    account.close();
+
+    REQUIRE_THROWS_AS(account.deposit(50), std::logic_error);
+}
+
+TEST_CASE("withdraw_into_closed_account_throws_error") {
```

Comment:
"withdraw_from" instead of "withdraw_into"
(Disclaimer: I'm not a native English speaker.)